src/main/scala/redis/Sentinel.scala

```diff
@@ -138,6 +178,23 @@ abstract class SentinelMonitored(system: ActorSystem) {
 
     Await.result(ff, 15 seconds)
   }
+
+  def withSlavesAddr[T](initFunction: Seq[(String, Int)] => T): T = {
+     import scala.concurrent.duration._
+
+    val fslaves = Future.sequence(sentinelClients.values.map(_.slaves(master)))
+      .map { lm =>
+        val ipAndPort = lm.flatten.flatMap { m =>
+          (m.get("ip"), m.get("port")) match {
+            case (Some(ip), Some(port)) => Some((ip, port.toInt))
+            case _ => None
+          }
+        }.toSeq.distinct
+        initFunction(ipAndPort)
+      }
```

Comment:
val fslaves = Future.sequence(sentinelClients.values.map(_.slaves(master)))
      .map { lm =>
        val ipPortBuilder = Set.newBuilder[(String, Int)]
        for {
          slaves <- lm
          slave <- slaves
          ip <- slave.get("ip")
          port <- slave.get("port")
        } yield {
          ipPortBuilder += ip -> port.toInt
        }
        initFunction(ipPortBuilder.result().toSeq)
      }